src/platform/i386/gdt.c

```diff
@@ -26,25 +27,25 @@ chal_tls_update(vaddr_t addr)
  * user-mode selectors or a TSS, but we need both now.
  */
 void
-gdt_init(void)
+gdt_init(const u32_t cpu_id)
 {
 	u64_t gdtr_operand;
 
 	/* Initialize GDT. */
-	gdt[SEL_NULL / sizeof *gdt]  = 0;
-	gdt[SEL_KCSEG / sizeof *gdt] = make_code_desc(0);
-	gdt[SEL_KDSEG / sizeof *gdt] = make_data_desc(0);
-	gdt[SEL_UCSEG / sizeof *gdt] = make_code_desc(3);
-	gdt[SEL_UDSEG / sizeof *gdt] = make_data_desc(3);
-	gdt[SEL_TSS / sizeof *gdt]   = make_tss_desc(&tss);
-	gdt[SEL_UGSEG / sizeof *gdt] = make_data_desc(3);
+	gdt[cpu_id][SEL_NULL / sizeof *(gdt[cpu_id])]  = 0;
```

Comment:
Would it be better to have a `volatile u64_t *core_gdt[]` set to `gdt[cpu_id]` and use that? Or a wrapper that does that? We probably don't touch this code once it's working, so may not matter as much.

Reply:
I will keep this based on your points.